src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,45 @@
+import { useCallback } from 'react';
+import cn from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+
+const arrayOfCenturies = ['16', '17', '18', '19', '20'];
+const sexParams = [{ All: null }, { Male: 'm' }, { Female: 'f' }];
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || null;
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+
+  const onQueryChange
+  = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
```suggestion
  const handleQueryChange = useCallback((
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
```
Fix formatting. Also, it would be great to name handlers according to these rules
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364